### ISSUE TITLE
Torch distributions rewrite. 

### ISSUE THREAD

**Issue by srush** [harvardnlp/namedtensor#81]
From conversations with @justinchiu 

The current distributions framework need a bit of an overhaul. We think this is an area with a lot of potential, so we are thinking through the constructors and use cases.

**Comment by srush** [comment]
Mostly done with https://github.com/harvardnlp/namedtensor/pull/88

**Comment by Ricocotam** [comment]
Hi,
May I suggest adding some documentations on the distribution module ?

Or could you give some hints on how it works so I could do it ? I tried to use it but there's no documentation and the code isn't easy to understand.

**Comment by srush** [comment]
Yeah, there will be a writeup with clear description, but it is still being worked out.

**Comment by Ricocotam** [comment]
Alright, may I help ?
